model/task/task.go

```diff
@@ -1946,7 +1946,9 @@ func (t *Task) displayTaskPriority() int {
 func (t *Task) Reset() error {
 	return UpdateOne(
 		bson.M{
-			IdKey: t.Id,
+			IdKey:       t.Id,
+			StatusKey:   bson.M{"$in": evergreen.TaskCompletedStatuses},
+			ArchivedKey: true,
```

Comment:
I'm a bit uncertain why Archived is included in this reset. Wouldn't we only want to reset if it _hasn't_ been archived yet?

Reply:
Archived is only set on the old task collection I thought. I'm not certain how this would ever hit something since this is for the current collection so I'm assuming I'm missing some context.

Reply:
This is necessary for idempotency since we only want to reset the current execution if it was preceded by an archive: https://github.com/evergreen-ci/evergreen/pull/5816#discussion_r933348376

Reply:
This logic all makes sense -- my concern with reusing Archived is that right now we use that field to understand if something is something is in the old_tasks collection or not (for example, here https://github.com/evergreen-ci/evergreen/blob/c8ac8688404d6dd1ac0e420e206238ef5895b98e/model/task/task.go#L2651 we use OldTaskId if Archived is set). 

Should we start using OldTaskId to indicate if something is in the old_tasks collection instead? i.e. change existing usage of `Archived == true` to `OldTaskId != ""` ? This may also limit confusion with this field meaning multiple things.

Reply:
I agree about not wanting multiple meaning, but I think `OldTaskId` is only set for the task execution in the old task collection and not for the latest task execution in the regular task collection (i.e. the task doc we want to reset). We want a flag on the latest task indicating [this update](https://github.com/evergreen-ci/evergreen/blob/c8ac8688404d6dd1ac0e420e206238ef5895b98e/model/task/task.go#L2509-L2521) succeeded so that the latest task is in a "waiting to reset" state.

Reply:
Although that solution makes sense I feel that we can keep this solution simpler without introducing cross collection lookups and aggregations. We'd also need to ensure that the execution incrementation and variable unsetting on the task in the second update linked above is handled properly if we remove the DB update altogether.  

Since the multiple meaning for `Archived` was the initial concern I've added a new single bool `CanReset` on the task doc to indicate if it's in a safe state to reset. This is now replacing all of the usages of Archived we were using in this PR.

Reply:
Sorry yeah I wasn't suggesting we use OldTaskId to handle the stuff in this PR but rather to replace the _old_ use of Archived to check whether a document was in the tasks or old_tasks collection.